### PR TITLE
fix(chunker): reject negative token overlap

### PR DESCRIPTION
**Pull request by Shizoqua**
## Description

`_window_step()` calculates the fixed-token chunker's stride as `chunk_token_size - chunk_overlap_token_size`.

When `chunk_overlap_token_size` is negative, the stride becomes larger than the configured chunk size. This silently skips tokens between windows instead of producing the intended overlap.

[PR #3396](https://github.com/HKUDS/LightRAG/pull/3396) added upper-bound validation for overlap values greater than or equal to the chunk size. This change completes the boundary validation by rejecting negative values at the same shared stride calculation point.

## Related Issues

No related issue.

## Changes Made

- Reject negative `chunk_overlap_token_size` values with `ValueError`.
- Add a regression test confirming negative overlap is rejected.
- Add control coverage confirming zero and positive overlap still produce complete, gap-free token coverage.

The validation is placed in `_window_step()` because it is the shared enforcement point for fixed-token window stride calculation.

## Checklist

- [x] Changes tested locally
- [x] Code reviewed
- [x] Documentation updated (if necessary)
- [x] Unit tests added (if applicable)

## Additional Notes

Verified against `main` at `d692105`. The regression test fails on the unmodified code and passes after the fix.

The focused boundary tests pass, the chunker test suite passes excluding three pre-existing network-dependent tests, and `pre-commit run --all-files` and `git diff --check` complete successfully.

### CODE REVIEW

**Comment by danielaskdd** [comment]
@codex review

**Comment by danielaskdd** [comment]
@codex review

File: lightrag/chunker/token_size.py [hunk 1]
```diff
@@ -118,12 +118,21 @@ def _window_step(chunk_token_size: int, chunk_overlap_token_size: int) -> int:
...
     """
     if chunk_overlap_token_size >= chunk_token_size:
         raise ValueError(
             f"chunk_overlap_token_size ({chunk_overlap_token_size}) must be < "
             f"chunk_token_size ({chunk_token_size})"
         )
+    if chunk_overlap_token_size < 0:
+        raise ValueError(
+            f"chunk_overlap_token_size ({chunk_overlap_token_size}) must be >= 0"
+        )
```
[the remark this answers is not in the document]
Comment (Shizoqua) [reply lightrag/chunker/token_size.py]: Fixed in 15788b3. Negative overlap is now validated at the start of chunking_by_token_size(), before delimiter handling, so short segments and split_by_character_only=True are covered. The existing upper-bound validation remains lazy because it only applies when token windows are created. Regression coverage was added for both split modes.
